src/parser/jobs/ast/modules/Sect/Sect.tsx

```diff
@@ -1,163 +1,90 @@
-				aspectedCast = event
-
-			} else if (aspectedCast
-				&& (event.type === 'applybuff' || event.type === 'refreshbuff') && [...NOCTURNAL_SECT_STATUSES, ...DIURNAL_SECT_STATUSES].includes(event.ability.guid)) {
-				// This is an applybuff event of a sect buff that came after an aspected action
-
-				if (this.mapCastToBuff(aspectedCast.ability.guid).includes(event.ability.guid)
-					&& [...DIURNAL_SECT_STATUSES, ...NOCTURNAL_SECT_STATUSES].includes(event.ability.guid)) {
-
-					const SECT_ABILITY = DIURNAL_SECT_STATUSES.includes(event.ability.guid) ? DIURNAL_SECT_BUFF_ABILITY : NOCTURNAL_SECT_BUFF_ABILITY
-
-					// Fab a sect buff event at the start of the fight
-					events.splice(0, 0, {
-						...event,
-						timestamp: startTime - 1,
-						type: 'applybuff',
-						ability: SECT_ABILITY,
-						targetID: event.sourceID,
-						targetIsFriendly: true,
-					})
-					break
-				}
-			} else {
-				continue
-			}
+	override initialise() {
 
-		}
+		this.sectStatuses = SECT_STATUSES.map(statusKey => this.data.statuses[statusKey].id)
+		this.diurnalSectStatuses = DIURNAL_SECT_STATUSES.map(statusKey => this.data.statuses[statusKey].id)
+		this.nocturnalSectStatuses = NOCTURNAL_SECT_STATUSES.map(statusKey => this.data.statuses[statusKey].id)
 
-		return events
+		//this section uses hooks to go through the various functions
+		this.addEventHook(filter<Event>()
+			.source(this.parser.actor.id)
+			.type('statusApply')
+			.status(oneOf([...this.sectStatuses, ...this.nocturnalSectStatuses, ...this.diurnalSectStatuses]))
+		, this.onApplyStatus)
+		this.addEventHook('complete', this.onComplete)
 	}
 
-	private onCast(event: CastEvent) {
-		// If they used a sect after the fight started, it means they pulled without one on; otherwise, we probably added it in normalize.
-		this.pullWithoutSect = event.timestamp >= this.parser.fight.start_time
+	//to look for the sect buff or any relating sects statuses we would expect
+	private onApplyStatus(event: Events['statusApply']) {
 
-		const sect = getDataBy(ACTIONS, 'id', event.ability.guid)
-		if (sect) {
-			this.activeSectId = sect.id
+		//if they switched it mid-fight, this section will pick it up
+		if (this.sectStatuses.includes(event.status)) {
+			if (this.activeSectId !== 0) {
+				//specifically used to check whether a sect status existed by previous status as noted below. if it didn't exist before this status, then we assume they didn't have it to begin with since sect cannot be changed in combat
+				//note: this will pick up instances where the sect is changed moments before the prepull as the sect takes approx 3.5 seconds to be applied; the sect will at least take 1 oGCD plus risks not being prepared for the pull.
+				this.pullWithoutSect = true
+			}
+			this.activeSectId = event.status === this.data.statuses.DIURNAL_SECT.id ? this.data.actions.DIURNAL_SECT.id
+				: this.data.actions.NOCTURNAL_SECT.id
```

Comment:
Mixing and matching actions and statuses isn't a good idea. Since you know what statuses are coming in, setting it to event.status alone is good enough.

Reply:
original code used actions. must've gotten confused with Status['id'] when I should've used Action['id]
either way, I just updated all to statuses except icons for stylistic purposes